pkg/client/interfaces.go

```diff
@@ -50,7 +50,7 @@ type Reader interface {
 	// Get retrieves an obj for the given object key from the Kubernetes Cluster.
 	// obj must be a struct pointer so that obj can be updated with the response
 	// returned by the Server.
-	Get(ctx context.Context, key ObjectKey, obj Object) error
+	Get(ctx context.Context, key ObjectKey, obj Object, opts ...GetOption) error
```

Comment:
Generally the change looks good to me, but it needs to be marked as breaking since we change the interface here. It is very likely that ppl have fake implementations of the client that need changing to adhere to the new signature

Reply:
Sure, I have updated the title. Do you think it is necessary to add the breaking warning into code comments?